core/src/main/java/com/linecorp/armeria/server/ShutdownSupport.java

```diff
@@ -35,31 +35,42 @@ static ShutdownSupport of(AccessLogWriter accessLogWriter) {
         });
     }
 
-    static ShutdownSupport of(ScheduledExecutorService executor) {
+    static ShutdownSupport of(ExecutorService executor) {
         requireNonNull(executor, "executor");
         return () -> {
-            final ScheduledExecutorService e;
-            if (executor instanceof UnstoppableScheduledExecutorService) {
-                e = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
-            } else {
-                e = executor;
+            executor.shutdown();
+            try {
+                while (!executor.isTerminated()) {
+                    executor.awaitTermination(1, TimeUnit.HOURS);
+                }
+            } catch (InterruptedException cause) {
+                logger.warn("During the termination wait, an interrupt occurs, attempting to forcefully " +
+                            "terminate the {}:", executor, cause);
+                executor.shutdownNow();
+                Thread.currentThread().interrupt();
             }
```

Comment:
I think it's probably safer to wait for the executor termination so that the server is cleanly shut down.
i.e. If long running tasks are executing a transaction, I doubt users would want armeria to forcefully cancel the task.

> Therefore, the cancellation policies of the two codes are different.

I think it would be possible to match the two behavior such that:
- `shutdownWhenStopping(executor)`: doesn't terminate until shutdown is completed
- `shutdownWhenStopping(executor, millis)`: doesn't terminate until either 1) shutdown is completed 2) millis is passed

Interruption won't have an effect

Reply:
Thank you for the explanation! As you mentioned, users may not want to cancel ongoing tasks. I will revert the code to the previous logic for indefinite waiting.